clap_complete/tests/testsuite/dynamic.rs

```diff
@@ -592,9 +593,32 @@ val3
 
 #[test]
 fn suggest_custom_arg_value() {
-    let mut cmd = Command::new("dynamic").arg(clap::Arg::new("custom").long("custom"));
+    #[derive(Debug)]
+    struct MyCustomCompleter {}
+
+    impl CustomCompleter for MyCustomCompleter {
+        fn custom_hint(&self) -> Vec<CompletionCandidate> {
+            vec![
+                CompletionCandidate::new("custom1").visible(true),
+                CompletionCandidate::new("custom2").visible(true),
+                CompletionCandidate::new("custom3").visible(true),
+            ]
+        }
+    }
+
+    let mut arg = clap::Arg::new("custom").long("custom");
+    arg.add::<ArgValueCompleter>(ArgValueCompleter::new(MyCustomCompleter {}));
+
+    let mut cmd = Command::new("dynamic").arg(arg);
```

Comment:
Looks like `add` had the wrong return type.  That should now be fixed and you can now chain it with the other `Arg` methods